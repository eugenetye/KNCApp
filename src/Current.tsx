import { View, Text } from 'react-native'
import React from 'react'
import { useFonts } from 'expo-font';

export const Current = () => {
  const [fontsLoaded] = useFonts({
    'Questrial-Regular': require('../assets/fonts/Questrial-Regular.ttf'),
  });
  if (!fontsLoaded) {
    return null;
  }

  return (
    <View>
      <Text style={{ fontFamily: 'Questrial-Regular', fontSize: 40, padding:15 }}>Current</Text>
    </View>
  )
}

  