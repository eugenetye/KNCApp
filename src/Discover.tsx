import { ScrollView, Text, View, Pressable, Image, StyleSheet } from 'react-native'
import React, { useState, useLayoutEffect, useEffect, Component } from 'react'
import { useFonts } from 'expo-font';
import { useNavigation } from '@react-navigation/native';
import { FIRESTORE_DB, FIREBASE_STORAGE } from '../firebaseConfig';
import { collection, onSnapshot } from 'firebase/firestore';
import CachedImage from '../components/CachedImage';
import { StackNavigationProp } from '@react-navigation/stack';

export type RootStackParamList = {
  Info_Template: { param: any } | undefined;
}

const styles = StyleSheet.create({
  image: {
    marginTop: -20,
    alignItems: 'center', 
    justifyContent: 'center',
  },
  button: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#046339',
    borderRadius: 15  ,
    padding: 10,
    height: 50,
    width: "80%",
    marginBottom: 10,
  }
});

export const Discover = () => {
  const [fontsLoaded] = useFonts({
    'Questrial-Regular': require('../assets/fonts/Questrial-Regular.ttf'),
  });

  const [datas, setDatas] = useState<any[]>([]); // Initialize with an empty array
  const navigation = useNavigation<StackNavigationProp<RootStackParamList>>();

  useEffect(() => {
    const querySnapshot = collection(FIRESTORE_DB, "discover");

    const subscriber = onSnapshot(querySnapshot, {
      next: (snapshot) => {
        
        const datas: any[] = [];
        snapshot.docs.forEach((doc) => {
          datas.push({
            id: doc.id,
            ...doc.data(),
          })
        });
        setDatas(datas);
      },
    });

    return () => subscriber();
  }, []);

  if (fontsLoaded) {
    return (
      <ScrollView>
        <View>
        <View>
            <Text style={{ fontFamily: 'Questrial-Regular', fontSize: 40, padding:15 }}>Discover</Text>
          </View>
          <View style={{ 
            paddingVertical: 12, backgroundColor: "#F8F8F8",
            borderRadius: 8,
            padding: 10,
            flexDirection: "column",
            alignItems: "flex-start",
            justifyContent: "space-between",
            margin: 14, }}>

            <CachedImage className='w-full h-40 object-cover rounded-2xl mt-3 mb-4' url = {'/discover/discover.jpeg'}
            />
          </View>
          <View style={{alignItems: 'center'}}>
            {datas.map((data, i) => (
                <Pressable style={styles.button} key={i} onPress={() => navigation.navigate('Info_Template', { param: data })}>
                <Text style={{color: 'white', fontSize: 20, fontFamily: 'Questrial-Regular'}}>{data.title}</Text>
                </Pressable>
              ))}       
          </View>

        </View>
      </ScrollView>
    )
  } else {
    return null;
  }

}
