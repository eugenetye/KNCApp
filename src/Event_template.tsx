import {
  ScrollView,
  Text,
  View,
  Pressable,
  StyleSheet,
  Dimensions,
} from "react-native";
import React, { useEffect, useLayoutEffect, useState } from "react";
import { useNavigation } from "@react-navigation/native";
import { FIREBASE_STORAGE } from "../firebaseConfig";
import { ref, listAll } from "firebase/storage";
import Ionicons from "@expo/vector-icons/Ionicons";
import { SafeAreaView } from "react-native-safe-area-context";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { CachedImage } from "../components/CachedImage";
import Carousel from "react-native-snap-carousel";
import { Linking } from "react-native";

const { height, width } = Dimensions.get("window");

const styles = StyleSheet.create({
  itemContainer: {
    flex: 1, // this will make the container expand to fill available space
    alignItems: "center",
    backgroundColor: "white",
    borderRadius: 8,
    padding: 20,
    justifyContent: "flex-start", // changed from 'center' to 'flex-start' to align children to the top
    marginTop: 60, // adjust this value to shift the container down under the back arrow
  },

  itemImg: {
    width: width * 0.7, // 80% of screen width
    height: height * 0.3, // 30% of screen height
    borderRadius: 8, // Added rounding to images
  },

  itemTitle: {
    fontSize: 24,
    fontWeight: "bold",
    marginTop: 10, // Space between the image and title
  },

  itemTitle2: {
    fontSize: 17, // You can customize these values
    fontWeight: "bold",
    color: "#333333", // Example color
    marginTop: 5, // Adjust spacing as needed
    textAlign: "center", // Customize alignment
    // Add other styling properties as needed
  },

  itemBody: {
    fontSize: 15,
    lineHeight: 20,
    marginTop: 10, // Space between the title and body
    textAlign: "center", // Center the text if it's a short sentence
  },

  safeArea: {
    flex: 1,
    backgroundColor: "#F8F8F8", // or any other background color you prefer
  },

  scrollView: {
    flex: 1,
  },

  viewContainer: {
    flex: 1,
  },

  backButton: {
    position: "absolute",
    top: 10, // safe distance from the top edge
    left: 10,
    width: 50,
    height: 50,
    borderRadius: 25,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "white",
    zIndex: 1,
  },
});

const button = StyleSheet.create({
  header: {
    alignItems: 'flex-start',
  },
  image: {
    marginTop: -20,
    alignItems: 'center',
    justifyContent: 'center',
  },
  button: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#046339',
    borderRadius: 15,
    padding: 10,
    height: 50,
    width: "100%",
    marginBottom: 10,
  },
  text: {
    color: 'white',
    fontSize: 20,
    fontFamily: 'Questrial-Regular',
  },
  headertext: {
    fontFamily: 'Questrial-Regular',
    fontSize: 40,
    padding: 15,
  }
});

export type RootStackParamList = {
  Past: { param?: any } | undefined;
  Current: { param?: any } | undefined;
  Event_template: { param?: any } | undefined;
  Event_item: { param?: any } | undefined;
};

const TextAndPicture = (route: any) => {
  const navigation = useNavigation<NativeStackNavigationProp<RootStackParamList>>();

  const item = route.params.param;

  const [images, setImages] = useState<any[]>([]);

  useEffect(() => {
    const fetchImages = async () => {
      const folderRef = ref(FIREBASE_STORAGE, item.type + "/fire_story"); // Assuming item.type is the folder name

      const listResult = await listAll(folderRef);

      const imageArray = await Promise.all(
        listResult.items.map((obj, i) => {
          return {
            //title: imageRef.name,
            body: item.bio[i], // Existing description
            imgUrl: obj.fullPath,
            header1: item.header1[i], // Assuming these are arrays and have the same length as bio
            header2: item.header2[i],
          };
        })
      ); 

      setImages(imageArray);
    };

    fetchImages();
  }, []);

  const { width: screenWidth } = Dimensions.get("window");
  const sliderWidth = screenWidth;
  const itemWidth = screenWidth * 0.8;

  interface YourItemType {
    imgUrl: string;
    title: string;
    body: string;
    header1: string;
    header2: string;
  }

  const renderItem = ({ item }: { item: any }) => (
    <View style={styles.itemContainer}>
      <CachedImage url={'/' + item.imgUrl} style={styles.itemImg} />

      <Text style={styles.itemTitle}>{item.header1}</Text>
      <Text style={styles.itemTitle2}>{item.header2}</Text>
      <Text style={styles.itemBody}>{item.body}</Text>
    </View>
  );

  useLayoutEffect(() => {
    navigation.setOptions({
      headerShown: false,
    });
  }, []);

  return (
    <SafeAreaView className="flex-1 relative">
      <ScrollView className="flex-1 px-4 py-6">
        <View className="relative">
          {/* Back button with higher zIndex */}

          <View
            className="absolute flex-row inset-x-0 justify-between"
            style={{ zIndex: 1 }}
          >
            <Pressable
              onPress={() => navigation.goBack()}
              className="w-10 h-10 rounded-md items-center justify-center bg-white"
            >
              <Ionicons size={24} name="arrow-back" />
            </Pressable>
          </View>

          {/* Picture or Carousel */}

          <Carousel
            layout="default"
            data={images}
            renderItem={renderItem}
            sliderWidth={width}
            itemWidth={width * 0.8}
            vertical={false}
          />
          <Pressable 
            style={button.button} 
            className='mt-10 mb-11' 
            onPress={() => {
              Linking.openURL('https://storymaps.com/stories/09ed079e0c49449fa0f363f0d87b052c')
                .catch(err => console.error('An error occurred', err));
            }}
          >
            <Text style={button.text}>Access the Story Map Here</Text>
          </Pressable>
          <View style={{ height: 80 }} />
        </View>
      </ScrollView>
    </SafeAreaView>
  );
};

const Event_template = ({ route }: any) => {
  const item = route.params.param;

  return (
    <TextAndPicture {...route} />
  );
};

export default Event_template;
