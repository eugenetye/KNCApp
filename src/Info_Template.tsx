import { ScrollView, Text, View, Pressable, Image, StyleSheet } from 'react-native'
import React, { useEffect, useLayoutEffect, useState } from 'react'
import { useNavigation } from '@react-navigation/native';
import { FIREBASE_STORAGE } from "../firebaseConfig";
import { ref, getDownloadURL, listAll } from "firebase/storage";
import Ionicons from "@expo/vector-icons/Ionicons";
import { SafeAreaView } from 'react-native-safe-area-context';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import AudioPlayer from '../components/AudioPlayer';

const styles = StyleSheet.create({
  header: {
    alignItems: 'flex-start', 
  },  
  image: {
    marginTop: -20,
    alignItems: 'center', 
    justifyContent: 'center',
  },
  button: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#046339',
    borderRadius: 15  ,
    padding: 10,
    height: 50,
    width: "100%",
    marginBottom: 10,
  },
  text: {
    color: 'white',
    fontSize: 20,
    fontFamily: 'Questrial-Regular',
  },
  headertext: {
    fontFamily: 'Questrial-Regular', 
    fontSize: 40, 
    padding: 15,
  }
});

const setImageLink = (uid: string, type: string) => {
  let pages: string[] = ['discover', 'current', 'past', 'trails']; 

  for (let i = 0; i < pages.length; i++) {
    if (type == pages[i]) {
      return ('/' + pages[i] + '/' + uid + '.jpeg');
    }
  }
}

export type RootStackParamList = {
  Trails: { param?: any } | undefined;
};

const AudioPlay = ({ file }: { file: string}) => {
  const navigation = useNavigation<NativeStackNavigationProp<RootStackParamList>>();

  return (
  <SafeAreaView className='flex-1 relative'>
  <ScrollView className='flex-1 px-4 py-6'>
    <View className='relative'>
      <View className='absolute flex-row inset-x-0 justify-between '>
        <Pressable onPress={() => navigation.goBack()}
          className='w-10 h-10 rounded-md items-center justify-center bg-white'>
          <Ionicons size={24} name="arrow-back"></Ionicons>
        </Pressable>
      </View>
      <AudioPlayer file={file}/>
    </View>
  </ScrollView>
</SafeAreaView>
  )
}

const TextAndPicture = ({ route }: any) => {
  const navigation = useNavigation<NativeStackNavigationProp<RootStackParamList>>();

  const item = route.params.param
  const imgLink = setImageLink(item.uid, item.type);

  const [url, setUrl] = useState(imgLink);

  useEffect(() => {
      const func = async () => {
      const reference = ref(FIREBASE_STORAGE, imgLink); 
      await getDownloadURL(reference).then((x) => {
          setUrl(x);
      })
  }

      func();
  }, []);

  
  
  useLayoutEffect(() => {
      navigation.setOptions({
        headerShown: false,
      });
  }, []);

  const showButton = () => {
      if (item.type === "trails") {
          return (
            <Pressable style={styles.button} onPress={() => alert('Button pressed')}>
            <Text style={styles.text}>Access the trail map here</Text>
          </Pressable>  
          )
      }
      else {
          return (<Text disabled={true}/>)
      }
  }

  return (
  <SafeAreaView className='flex-1 relative'>
  <ScrollView className='flex-1 px-4 py-6'>
    <View className='relative'>

      <View className='absolute flex-row inset-x-0 justify-between '>
          <Pressable onPress={() => navigation.goBack()}
          className='w-10 h-10 rounded-md items-center justify-center bg-white'>
              <Ionicons size={24} name="arrow-back"></Ionicons>
          </Pressable>
      </View>

      <View className='py-5 mt-9'>
        <Text
          style={{
            fontSize: 27,
            fontWeight: "500",
            fontFamily: 'Questrial-Regular',
            color: '#1b747c',
          }}
        >
          {item.name}
        </Text>
      </View>
      <View>
      <Image className='w-full h-40 object-cover rounded-2xl mt-3'
          source={{ uri: url }}
      />
      <View className='py-5 mt-3'>
        <Text
          style={{
            fontSize: 19,
            fontWeight: "500",
            fontFamily: 'Questrial-Regular',
          }}
        >
          {item.description}
        </Text>
      </View>
      {showButton()}
      </View>
        </View>
      </ScrollView>
    </SafeAreaView>
  )
}

const Info_Template = ({ route }: any) => {
  const item = route.params.param;

  const file_name: string = item.files[0];
  return (item.audio
    ? <AudioPlay file={file_name} />
    : <TextAndPicture {... route}/>)
}


export default Info_Template
