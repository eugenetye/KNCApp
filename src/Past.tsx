import { View, Text, ScrollView } from 'react-native'
import React, {useEffect, useState} from 'react'
import { useFonts } from 'expo-font';
import { FIRESTORE_DB } from '../firebaseConfig';
import { collection, onSnapshot } from 'firebase/firestore';
import { Event_item } from '../components/Event_item';

export const Past = () => {
  const [fontsLoaded] = useFonts({
    'Questrial-Regular': require('../assets/fonts/Questrial-Regular.ttf'),
  });


  const [datas, setDatas] = useState<any[]>([]); // Initialize with an empty array


  useEffect(() => {
    const querySnapshot = collection(FIRESTORE_DB, "past");

    const subscriber = onSnapshot(querySnapshot, {
      next: (snapshot) => {
        const datas: any[] = [];
        snapshot.docs.forEach((doc) => {
          datas.push({
            id: doc.id,
            ...doc.data(),
          })
        });
        setDatas(datas);
      },
    });

    return () => subscriber();
  }, []);
  
  return (
    <ScrollView>
      <View>
        <View>
          <Text style={{ fontFamily: 'Questrial-Regular', fontSize: 40, padding:15 }}>Past Exhibits</Text>
        </View>

        {datas.map((data, i) => (
            <Event_item key={i} item={data}/> 
          ))}

      </View>
    </ScrollView>
  )
}