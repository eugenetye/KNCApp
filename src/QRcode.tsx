import { View, Text } from 'react-native'
import React, { useEffect } from 'react'
import { useFonts } from 'expo-font';
import { BarCodeScanner } from 'expo-barcode-scanner';
import * as FileSys from 'expo-file-system';
import { AVPlaybackStatus, Audio } from 'expo-av';
import AudioPlayer from '../components/AudioPlayer';
import { FIREBASE_STORAGE } from "../firebaseConfig";
import { ref, getDownloadURL } from "firebase/storage";

type QRProps = {
  navigation: any
};
type QRState = {
  hasCameraPermission: Boolean,
  audioFile: string,
};

const SAVE_DIR = FileSys.cacheDirectory + 'kncapp/';

async function ensureDirExists() {
  await FileSys.makeDirectoryAsync(SAVE_DIR, { intermediates: true });
}

async function saveDirExists(): Promise<Boolean> {
  return (await FileSys.getInfoAsync(SAVE_DIR)).exists;
}

async function saveFileExists(file: string): Promise<Boolean> {
  return (await FileSys.getInfoAsync(file)).exists;
}

export default class QRScanner extends React.Component<QRProps, QRState> {
  state: QRState = {
    hasCameraPermission: false,
    audioFile: '',
  };

  componentDidMount() {
    this._requestCameraPermission();
  }

  _requestCameraPermission = async () => {
    const { status } = await BarCodeScanner.requestPermissionsAsync();
    this.setState({
      hasCameraPermission: status === 'granted',
    });
  };

  _handleBarCodeRead = async (result: any) => {
    var new_param = JSON.parse(result.data);

    // We only cache audio so far
    //
    // TODO: this will actually need to be a loop, the
    // first QR code we scan at the KNC building will start
    // downloading all the audio files and pictures if there are
    // any.
    if (new_param.audio) {
      try {
        if (!(await saveDirExists())) {
          await ensureDirExists();
        }
        for (let i = 0; i < new_param.files.length; i++) {
          const file = new_param.files[i];
          if (!(await saveFileExists(SAVE_DIR + file + '.mp3'))) {
            const reference = ref(FIREBASE_STORAGE, file + '.mp3');
            const url = await getDownloadURL(reference);
            await FileSys.downloadAsync(url, SAVE_DIR + file + '.mp3');
          }
          new_param.files[i] = SAVE_DIR + file + '.mp3';
        }
      } catch (e) {
        console.log(`We got an error: ${e}`);
      }
    }

    // This happens for both audio and picture pages
    this.props.navigation.navigate('Info_Template', { param: new_param });
  };

  render() {
    return (
      <View>
        {this.state.hasCameraPermission === null ? (
          <Text>Requesting for camera permission</Text>
        ) : this.state.hasCameraPermission === false ? (
          <Text style={{ color: '#fff' }}>
            Camera permission is not granted
          </Text>
        ) : (
          <View
            style={{
              height: '75%',
              width: '100%'
            }}>
            <BarCodeScanner
              onBarCodeScanned={this._handleBarCodeRead}
              style={{
                height: '100%',
                width: '100%',
              }}
            />
          </View>)}
      </View>
    );
  }
}

export const QRcode = (props: any) => {
  return (
    <QRScanner {...props} />
  )
}

