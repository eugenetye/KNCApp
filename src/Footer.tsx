import { View, Text } from 'react-native'
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import React from 'react'
import Discover from './Discover'
import Trails from './Trails'
import Current from './Current';
import Past from './Past';

const Tab = createBottomTabNavigator();

const Footer = () => {
  return (
    <Tab.Navigator>
      <Tab.Screen name="Discover" component={Discover} />
      <Tab.Screen name="Trails" component={Trails} />
      <Tab.Screen name="Current" component={Current} />
      <Tab.Screen name="Past" component={Past} />
    </Tab.Navigator>
  );
}

export default Footer