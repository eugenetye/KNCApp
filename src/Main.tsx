import { View, Text, ScrollView } from 'react-native'
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import Ionicons from "@expo/vector-icons/Ionicons";
import React from 'react'
import {Trails} from './Trails'
import {Discover} from './Discover'
import {Current} from './Current';
import {Past} from './Past';
import {QRcode} from './QRcode';
import { RouteProp } from '@react-navigation/native'; // Import RouteProp
import { ParamListBase } from '@react-navigation/native'; // Import ParamListBase

const Tab = createBottomTabNavigator();

function showScreens(
  pageName: string,
  comp: () => React.JSX.Element | null
) {
  return (
    <Tab.Screen
      name={pageName}
      component={comp}
      options={{
        headerShown: false,
      }}
    />
  );
}

const discoverScreen = showScreens("Discover", Discover);
const trailsScreen = showScreens("Trails", Trails);
const currentScreen = showScreens("Current", Current);
const pastScreen = showScreens("Past", Past);

const ICONS: Record<string, string> = {
  "Discover": "navigate-circle-outline",
  "Trails": "map-outline",
  "Current": "newspaper-outline" ,
  "Past": "calendar-outline",
  "QRcode": "qr-code-outline",
};

export const Main = () => {
  return (
    <Tab.Navigator
      screenOptions={({ route, navigation }) => ({
        tabBarIcon: ({ focused, color, size }) => {
          const iconName: string = ICONS[route.name];
          return (
            <Ionicons
              name={iconName}
              size={25}
              color={focused ? "#C5DFC5" : "grey"}
            />
          );
        },
        tabBarActiveTintColor: "green",
        tabBarInactiveTintColor: "gray",
      })}
    >
      {discoverScreen}
      {trailsScreen}
      <Tab.Screen
        name={"QRcode"}
        options={{
          headerShown: false,
        }}
      >
        {({ route, navigation }) =>
          <QRcode navigation={navigation}/>
        }
      </Tab.Screen>
      {currentScreen}
      {pastScreen}
    </Tab.Navigator>
  );
};
