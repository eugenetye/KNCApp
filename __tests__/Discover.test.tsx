import React from 'react';
import { act, create, ReactTestRendererJSON } from 'react-test-renderer';

import { Discover } from '../src/Discover';

describe('<Discover />', () => {
    // Test Case ID = 9
    it('loads data', async () => {
        const comp = create(<Discover />);
        let tree = comp.toTree()?.instance;
        expect(tree?.state.datas.length).toEqual(0);
    });

    // Test Case ID = 10
    it('renders correctly', async () => {
        const comp = new Discover({});
        comp.state.fontsLoaded = true;
        const tree = comp.render();
        expect({ ...tree, children: null } as ReactTestRendererJSON)
            .toMatchSnapshot();
    });
});
