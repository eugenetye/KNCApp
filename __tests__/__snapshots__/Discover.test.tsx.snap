// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Discover /> renders correctly 1`] = `
<ScrollView>
  <View>
    <View>
      <Text
        style={
          {
            "fontFamily": "Questrial-Regular",
            "fontSize": 40,
            "padding": 15,
          }
        }
      >
        Discover
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "flex-start",
        "backgroundColor": "#F8F8F8",
        "borderRadius": 8,
        "flexDirection": "column",
        "justifyContent": "space-between",
        "margin": 14,
        "padding": 10,
        "paddingVertical": 12,
      }
    }
  >
    <CachedImage
      className="w-full h-40 object-cover rounded-2xl mt-3 mb-4"
      url="/discover/discover.jpeg"
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
      }
    }
  />
</ScrollView>
`;
